### PR TITLE
Option to overwrite exported variables in TorchDistributedEnvironment

### PR DESCRIPTION
**Pull request by qasfb**
This can be useful when launching a job from another job. Otherwise the environment variables are inherited, leading to a RuntimeError.

### CODE REVIEW

File: submitit/helpers.py [hunk 1]
```diff
@@ -352,7 +359,7 @@ def export(self, set_cuda_visible_devices: bool = True) -> "TorchDistributedEnvi
             "LOCAL_WORLD_SIZE": str(self.local_world_size),  # Not required
         }
         for key in env_vars:
-            if key in os.environ:
+            if key in os.environ and not overwrite:
                 raise RuntimeError(f"Cannot export environment variables as {key} is already set")
```
Comment (patricklabatut) [inline submitit/helpers.py:363]: Nit: Clearer (early check + non-negative condition):

```
if preserve_env_vars:
    for key in env_vars:
        if key in os.environ:
            raise RuntimeError(f"Cannot export environment variables as {key} is already set") 
```

**Comment by patricklabatut** [comment]
LGTM (see suggestion though) but hold until @jrapin and @gwenzek take a look.

**Comment by jrapin** [comment]
current CI issue should be fixed if you merge main inside your branch (I just pushed a fix)

**Review by jrapin — APPROVED** [review]
fine with it on my side

**Comment by qasfb** [comment]
I think the "overwrite" option is nice from a user perspective

File: submitit/helpers.py (hunk 1 above)
Comment (patricklabatut) [reply submitit/helpers.py:392]: Could the `overwrite` check still be moved higher?

**Pull request merged**